corehq/apps/app_manager/views/apps.py

```diff
@@ -793,3 +804,34 @@ def drop_user_case(request, domain, app_id):
         _('You have successfully removed User Case properties from this application.')
     )
     return back_to_main(request, domain, app_id=app_id)
+
+
+@require_GET
+@require_can_edit_apps
+def pull_master_app(request, domain, app_id):
+    app = get_current_app_doc(domain, app_id)
+    master_app = get_app(None, app['master'], latest=True)
+    params = {}
+    if app['domain'] in master_app.linked_whitelist:
+        excluded_fields = set(Application._meta_fields).union(
+            ['date_created', 'build_profiles', 'copy_history', 'copy_of', 'name', 'comment', 'doc_type']
+        )
+        master_json = master_app.to_json()
+        for key, value in master_json.iteritems():
+            if key not in excluded_fields:
+                app[key] = value
+        app['version'] = master_json['version']
+        wrapped_app = wrap_app(app)
+        wrapped_app.copy_attachments(master_app)
+        wrapped_app.save(increment_version=False)
+    return HttpResponseRedirect(reverse_util('view_app', params=params, args=[domain, app_id]))
```

Comment:
Would it make sense to throw an error the app's domain was not in the linked whitelist? It looks like this does nothing if that was the case.